dotnet/src/Connectors/Connectors.Ollama/Extensions/OllamaServiceCollectionExtensions.cs

```diff
@@ -110,10 +111,21 @@ public static IServiceCollection AddOllamaChatCompletion(
         Verify.NotNull(services);
 
         services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
-            new OllamaChatCompletionService(
-                modelId: modelId,
-                endpoint: endpoint,
-                loggerFactory: serviceProvider.GetService<ILoggerFactory>()));
+        {
+            var ollamaClient = new OllamaApiClient(endpoint, modelId);
+            var logger = serviceProvider.GetService<ILoggerFactory>()?.CreateLogger(ollamaClient.GetType());
+
+            var chatClientBuilder = new ChatClientBuilder()
+                .UseFunctionInvocation(config =>
+                    config.MaximumIterationsPerRequest = 128);
```

Comment:
Should this parameter be configurable? If not, then taking into account that it's used in multiple places, it's worth to define `128` as constant.

Reply:
Moved to a constant, similar to OpenAI.